velox/exec/MergeJoin.cpp

```diff
@@ -972,7 +1063,43 @@ RowVectorPtr MergeJoin::applyFilter(const RowVectorPtr& output) {
       if (!isAntiJoin(joinType_)) {
         rawIndices[numPassed++] = row;
 
-        if (!isRightJoin(joinType_)) {
+        if (isFullJoin(joinType_)) {
+          // Insert same row in next line and then apply nulls to left and
+          // right.
+          fullOuterOutput_ = BaseVector::create<RowVector>(
```

Comment:
I can't follow what is happening here. Can you expand with more comments explaining the logic?

Reply:
@pedroerp To handle rows that are filtered out during a full outer join. For these rows, it is necessary to insert additional data to ensure the result set is complete. Specifically, we need to generate two records: one record containing the columns from the left table along with nulls for the right table, and another record containing the columns from the right table along with nulls for the left table. 
For instance, the current output is filtered based on the condition t > 1. 

```
1, 1
2, 2
3, 3
```

In this scenario, we need to additionally insert a record 1, 1. Subsequently, we will set the values of the columns on the left to null and the values of the columns on the right to null as well. By doing so, we will obtain the final result set.

```
1,   null
null,  1
2, 2
3, 3
```